examples/reflection/convert.rs

```diff
@@ -0,0 +1,68 @@
+//! Illustrates how to do reflection with conversions
+
+use bevy::{prelude::*, reflect::DynamicStruct};
+
+fn main() {
+    App::new()
+        .register_type::<Foo>()
+        .add_systems(Startup, setup)
+        .run();
+}
+
+/// Deriving `Reflect` implements the relevant reflection traits. In this case, it implements the
+/// `Reflect` trait the `Struct` trait, and the `FromReflect` trait. `derive(Reflect)` assumes that
+/// all fields also implement Reflect.
```

Comment:
Nit: Technically, we only require `PartialReflect` (and `FromReflect` if deriving that). It might be easier for new users to just say `Reflect`, but maybe a compromise could be:
```suggestion
/// all fields also implement `PartialReflect`.
```